wallet/src/ui/app/redux/slices/transactions/index.ts

```diff
@@ -62,6 +62,51 @@ export const sendTokens = createAsyncThunk<
     }
 );
 
+type StakeTokensTXArgs = {
+    tokenTypeArg: string;
+    amount: bigint;
+};
+
+export const StakeTokens = createAsyncThunk<
+    TransactionResult,
+    StakeTokensTXArgs,
+    AppThunkConfig
+> (
+    'sui-objects/stake',
+    async (
+        { tokenTypeArg, amount },
+        { getState, extra: { api, keypairVault }, dispatch }
+    ) => {
+        const state = getState();
+        const coinType = Coin.getCoinTypeFromArg(tokenTypeArg);
+        const coins: SuiMoveObject[] = suiObjectsAdapterSelectors
```

Comment:
we could reuse the selector `accountCoinsSelector` from [here](https://github.com/MystenLabs/fastnft/blob/a7ec54bed3476de76c1d72779d343ea55261162e/wallet/src/ui/app/redux/slices/account/index.ts)